backend/src/meme/controllers/meme-controller.ts

```diff
@@ -0,0 +1,109 @@
+import {
+	Body,
+	Controller,
+	Get,
+	HttpException,
+	HttpStatus,
+	Param,
+	ParseIntPipe,
+	Post,
+	UploadedFile,
+	UseGuards,
+	UseInterceptors
+} from "@nestjs/common";
+import { JwtAuthGuard } from "../../auth/guards/jwt-auth.guard";
+import { MemeService } from "../services/meme.service";
+import { FileInterceptor } from "@nestjs/platform-express";
+import { MemeDto } from "../model/dto/meme.dto";
+@Controller("meme")
+class MemeController {
+	constructor(private memeService: MemeService) {}
+
+	@UseGuards(JwtAuthGuard)
+	@Get("random")
+	async getRandomMeme() {
+		const file = await this.memeService.findRandom();
+
+		if (!file) {
+			throw new HttpException("Not found", HttpStatus.NOT_FOUND);
+		}
+
+		return file;
+	}
+
+	@UseGuards(JwtAuthGuard)
+	@Post("file")
+	@UseInterceptors(FileInterceptor("file"))
+	async addMemeByFile(@UploadedFile() file) {
+		if (!file) {
+			throw new HttpException(
+				{
+					status: HttpStatus.BAD_REQUEST,
+					error: "No file given"
+				},
+				HttpStatus.BAD_REQUEST
+			);
+		}
+
+		const addFilePromises = await this.memeService.addMemeByFile(file);
+
+		try {
```

Comment:
IMO the best way to create try catch in <= ES6 is:

```ts
const resolved = await Promise.all(addFilePromises).catch(() => {throw Err});
```

Reply:
I personally like this way. But if You want i can change.